what-is-the-portal-network.md

```diff
@@ -0,0 +1,81 @@
+# The Portal Network
+
+## Through the DevP2P lens
+
+Existing Ethereum clients like Geth are built on top of the DevP2P protocol.
+The protocol is primarily designed to facilitate the transfer of data between
+full nodes.
+
+- Nodes on this network **must** be full nodes.
+- The network fundamentally assumes that all nodes have all data.
+
+The design of this network ends up dictating client design.  
+
+- Clients must fully sync the state and history.
+- Clients must build their own indices for looking up transactions by hash or blocks by number.
+
+This imposes significant limits on client design.
+
+## Through a different lens.
+
+If we ignore the limitations of the DevP2P network, and instead think about
+what clients could look like, here are a few things that come to mind.
+
+- Execution of blocks could be optional.
+- State database might only house portions of the state.
+- Elimination of sync times and only fetch data lazily as it is needed.
+- Reduced CPU/HDD/bandwidth usage
+
+Many of these things aren't feasible or realistic within the context of the
+DevP2P network.  They would require fundamental changes to the network design,
+many of which would likely reduce the overall security guarantees that the
+network provides and that client teams have chosen to prioritize.
+
+## So what is the Portal Network
+
+The Portal Network aims to build the networks that allow us to have a more
+diverse set of functionality in our choices for Ethereum clients.  The goal is
+not to replace the DevP2P network, but to provide supplimental networks that
+enable new functionality that isn't possible within the confines of the DevP2P
+network.
+
+- Special purpose networks that "do one thing well".
+  - Separate networks for history, state, mempool, etc.
+- Move the complexity out of the clients and into the network.
+  - Transaction index
+  - Block number index
+  - Mempool transaction validation
+- Ensure that network design is friendly towards resource constrained devices.
+  - No assumption of "full nodes"
+  - Data Radius mechanic allows nodes to tune their responsibility to match
+    their hardware resources.
+
+DevP2P can remain a simple and secure core component of Ethereum
+infrastructure, primarily used by clients that are focused on serving the
+protocol.  The Portal Network can then focus on delivering functionality that
+is focused on use cases of the users of Ethereum.
+
+## What happened to the original JSON-RPC API based Portal Network use case?
+
+The original pitch for the Portal Network was focused on delivering a network
+that allowed clients to the network to serve JSON-RPC requests by fetching data
+on-demand from the Portal Network.  Recently we've come to understand that this
+use case is likely to be difficult based on network latencies, which has forced
+us to re-evaluate our roadmap.
```

Comment:
A short example might help explain our reasoning on this. Let's say an `eth_call` makes 100 unique sloads, which costs 210,000 gas. At 0.5s for each storage lookup (which involves many network lookups to find the storage slot), that requires 50s to run the sloads in the call.

Reply:
Also, it's potentially worth noting exceptions.
- Academics & researchers that want to dive into a single transaction might not have any problem with that kind of wait. It could still seem significantly preferable to running a full node
- Maybe there are users who are read-only, and just want to see their token balances. Limited lookups like this could happen in a reasonable time.